### PR TITLE
Disable buttons for creating containers until they are needed

### LINKED ISSUE #1442: The 'Create' CTA on containers should be disabled at this stage
Here: 

screen shot 2016-03-07 at 12 23 55

### LINKED ISSUE #1443: 'Add container' should be disabled at this stage
Here: 

screen shot 2016-03-07 at 12 54 36

### PR DESCRIPTION
**Pull request by huwshimi**
Fixes #1443. Fixes #1442.

### CODE REVIEW

**Comment by huwshimi** [comment]
QA:
- open the machine view
- open the context menu at the top of the container colum
- "Add container" should be disabled
- add a machine
- "Add container" should now be enabled
- click "Add container"
- the "Create" button should be disabled
- choose a container type
- the "Create" button should now be enabled

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/4635/
Test PASSed.

File: jujugui/static/gui/src/app/components/machine-view/add-machine/add-machine.js [hunk 1]
```diff
@@ -247,13 +247,16 @@ YUI.add('machine-view-add-machine', function() {
...
         title: 'Cancel',
-        action: this.props.close
+        action: props.close
       }, {
-        title: this.props.unit ? 'Place' : 'Create',
+        title: props.unit ? 'Place' : 'Create',
         action: this._submitForm,
-        type: 'confirm'
+        type: 'confirm',
+        disabled:!props.unit && !props.machines && props.parentId &&
```
Comment (anthonydillon) [inline jujugui/static/gui/src/app/components/machine-view/add-machine/add-machine.js]: space after the colon?

**Comment by anthonydillon** [comment]
:+1: Nice improvements with such a small diff. Trivial comment in code.

File: jujugui/static/gui/src/app/components/machine-view/add-machine/add-machine.js (hunk 1 above)
Comment (anthonydillon) [inline jujugui/static/gui/src/app/components/machine-view/add-machine/add-machine.js]: Should you add a comment to this line to explain what your disabling based on. Its not clear from the conditions.

Comment (huwshimi) [inline jujugui/static/gui/src/app/components/machine-view/add-machine/add-machine.js]: Thanks @anthonydillon I've added the space and the comment.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/4652/
Test PASSed.

**Comment by kadams54** [comment]
Also :+1: on the code and QA is :ok:.

**Comment by hatched** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**